lib/reverse_words.rb

```diff
@@ -1,6 +1,37 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: The time complexity is O(n^2) where n is the length of my_words
+# Space complexity: The time complexity is O(n) where n is the length of the
+#                  string because no additional space is needed to store any
+#                  other strings or arrays
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  return nil if my_words.nil?
+
+  i = 0
+  j = 0
+  while j < my_words.length
+    if my_words[i] == " "
+      i += 1
+      j = i
+    end
+
+    while j < my_words.length && my_words[j] != " "
+      j += 1
+    end
+
+    reverse(my_words, i, j - 1)
+    i = j
+  end
+
```

Comment:
Good solution! When you start white boarding, I suggest using descriptive variable names in place of letters because it's easy to lose your place when nervous and a descriptive variable name handle the job of you trying to hold all of the data in your head. Also, in the Real World<sub>tm</sub> you should rarely use single letter variable names except for iterators in loops.